lang/python/pillow/patches/0001-setup.py-use-patch-from-Debian-and-tweak-it.patch

```diff
@@ -0,0 +1,316 @@
+-            for dirname in _find_library_dirs_ldconfig():
+-                _add_directory(library_dirs, dirname)
+-            if sys.platform.startswith("linux") and \
+-                    os.environ.get('ANDROID_ROOT', None):
+-                # termux support for android.
+-                # system libraries (zlib) are installed in /system/lib
+-                # headers are at $PREFIX/include
+-                # user libs are at $PREFIX/lib
+-                _add_directory(
+-                    library_dirs, os.path.join(os.environ["ANDROID_ROOT"], "lib")
+-                )
+-
+-        elif sys.platform.startswith("netbsd"):
+-            _add_directory(library_dirs, "/usr/pkg/lib")
+-            _add_directory(include_dirs, "/usr/pkg/include")
+-
+-        elif sys.platform.startswith("sunos5"):
+-            _add_directory(library_dirs, "/opt/local/lib")
+-            _add_directory(include_dirs, "/opt/local/include")
+-
+-        # FIXME: check /opt/stuff directories here?
+-
+-        # standard locations
+-        if not self.disable_platform_guessing:
+-            _add_directory(library_dirs, "/usr/local/lib")
+-            _add_directory(include_dirs, "/usr/local/include")
+-
+-            _add_directory(library_dirs, "/usr/lib")
+-            _add_directory(include_dirs, "/usr/include")
+-            # alpine, at least
+-            _add_directory(library_dirs, "/lib")
+-
+-        # on Windows, look for the OpenJPEG libraries in the location that
+-        # the official installer puts them
+-        if sys.platform == "win32":
+-            program_files = os.environ.get('ProgramFiles', '')
+-            best_version = (0, 0)
+-            best_path = None
+-            for name in os.listdir(program_files):
+-                if name.startswith('OpenJPEG '):
+-                    version = tuple(int(x) for x in
+-                                    name[9:].strip().split('.'))
+-                    if version > best_version:
+-                        best_version = version
+-                        best_path = os.path.join(program_files, name)
+-
+-            if best_path:
+-                _dbg('Adding %s to search list', best_path)
+-                _add_directory(library_dirs, os.path.join(best_path, 'lib'))
+-                _add_directory(include_dirs,
+-                               os.path.join(best_path, 'include'))
+ 
+         #
+         # insert new dirs *before* default libs, to avoid conflicts
+@@ -472,22 +404,10 @@ class pil_build_ext(build_ext):
+         feature = self.feature
+ 
+         if feature.want('zlib'):
+-            _dbg('Looking for zlib')
+-            if _find_include_file(self, "zlib.h"):
+-                if _find_library_file(self, "z"):
+-                    feature.zlib = "z"
+-                elif (sys.platform == "win32" and
+-                      _find_library_file(self, "zlib")):
+-                    feature.zlib = "zlib"  # alternative name
++            feature.zlib = "z"
```

Comment:
same comment here for leaving removed lines here

Reply:
I wonder if hard-coding `feature.zlib = "z"` (and the other lines for the other libraries) is really necessary. If the paths are correct, setup should be able to find the correct libraries. (I would again replace `sys.platform`s with `host_platform` as in the Debian patch.)

If it isn't able to find the libraries, we should find out why and fix that.